driver-core/src/test/java/com/datastax/driver/core/SpeculativeExecutionTest.java

```diff
@@ -168,6 +168,55 @@ public void should_start_speculative_execution_if_first_execution_takes_too_long
     }
 
     @Test(groups = "short")
+    public void should_start_speculative_execution_on_multiple_hosts_with_zero_delay() {
+        Cluster cluster = Cluster.builder()
+                .addContactPoints(scassandras.address(2).getAddress())
+                .withPort(scassandras.getBinaryPort())
+                .withLoadBalancingPolicy(loadBalancingPolicy)
+                .withSpeculativeExecutionPolicy(new ConstantSpeculativeExecutionPolicy(0, 2))
+                .withQueryOptions(new QueryOptions().setDefaultIdempotence(true))
+                .withRetryPolicy(new CustomRetryPolicy())
+                .withNettyOptions(nonQuietClusterCloseOptions)
+                .build();
+        Session session = cluster.connect();
+        host1 = TestUtils.findHost(cluster, 1);
+        host2 = TestUtils.findHost(cluster, 2);
+        host3 = TestUtils.findHost(cluster, 3);
+        errors = cluster.getMetrics().getErrorMetrics();
+
+        scassandras.node(1).primingClient().prime(PrimingRequest.queryBuilder()
+                .withQuery("mock query")
+                .withThen(then().withRows(row("result", "result1")).withFixedDelay(5L))
```

Comment:
I suggest raising your delays to a few seconds to make to sure the test will pass even in our CI environment (crazy scheduling behaviors happen there :) ).